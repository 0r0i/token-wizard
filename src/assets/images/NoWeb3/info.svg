<svg xmlns="http://www.w3.org/2000/svg" width="88" height="88">
    <path fill="#5C34A2" fill-rule="evenodd" d="M44 88C19.699 88 0 68.301 0 44S19.699 0 44 0s44 19.699 44 44-19.699 44-44 44zm0-84C21.909 4 4 21.909 4 44s17.909 40 40 40 40-17.909 40-40S66.091 4 44 4zm6 58H38a2 2 0 1 1 0-4h4V37h-4a2 2 0 1 1 0-4h6a2 2 0 0 1 2 2v23h4a2 2 0 1 1 0 4zm-6-33a2 2 0 1 1-.001-3.999A2 2 0 0 1 44 29z"/>
</svg>
