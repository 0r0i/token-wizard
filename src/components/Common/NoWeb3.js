import React, { Component } from 'react'

export default class NoWeb3 extends Component {
  render() {
    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Nifty wallet Not Found</h1>
              <p className="description">
                You don't have
                <a href="https://chrome.google.com/webstore/detail/nifty-wallet/jbdaocneiiinmjbjlgalhcelgbejmnid">
                  Nifty wallet
                </a>
                installed. Check Token Wizard GitHub for{' '}
                <a href="https://github.com/poanetwork/token-wizard" target="blank">
                  the instruction
                </a>.
              </p>
            </div>
          </div>
        </section>
      </div>
    )
  }
}
