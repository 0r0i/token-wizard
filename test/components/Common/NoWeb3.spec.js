import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'
import NoWeb3 from '../../../src/components/Common/NoWeb3'

configure({ adapter: new Adapter() })

describe('NoWeb3', () => {
  it('has a title, description and button container', () => {
    const wrapper = mount(<NoWeb3 />)

    expect(wrapper.find('.sw-NoWeb3_Title').exists()).toBeTruthy()
    expect(wrapper.find('.sw-NoWeb3_Description').exists()).toBeTruthy()
    expect(wrapper.find('.sw-NoWeb3_ButtonsContainer').exists()).toBeTruthy()
  })

  it('should go to URL', () => {
    const wrapper = shallow(<NoWeb3 />)

    global.open = jest.fn()
    wrapper.instance().goToURL('https://google.com')

    expect(global.open).toHaveBeenCalled()
  })

  it('should click the MetaMask button', () => {
    const wrapper = mount(<NoWeb3 />)
    const button = wrapper.find('.btn-ButtonMetamask')

    wrapper.instance().goToURL = jest.fn()
    button.simulate('click')

    expect(wrapper.instance().goToURL).toHaveBeenCalledTimes(1)
  })

  it('should click the NiftyWallet button', () => {
    const wrapper = mount(<NoWeb3 />)
    const button = wrapper.find('.btn-ButtonNiftyWallet')

    wrapper.instance().goToURL = jest.fn()
    button.simulate('click')

    expect(wrapper.instance().goToURL).toHaveBeenCalledTimes(1)
  })

  it(`should render ButtonNiftyWallet component`, () => {
    // Given
    const component = render.create(<NoWeb3 />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
