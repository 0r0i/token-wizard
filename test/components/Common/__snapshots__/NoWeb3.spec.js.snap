// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoWeb3 should render ButtonNiftyWallet component 1`] = `
<section
  className="sw-NoWeb3"
>
  <div
    className="sw-NoWeb3_Background background-blur"
  />
  <div
    className="sw-NoWeb3_Container"
  >
    <div
      className="sw-NoWeb3_Block"
    >
      <div
        className="sw-NoWeb3_InfoIcon"
      />
      <h1
        className="sw-NoWeb3_Title"
      >
        Wallet not found, or access to your Ethereum account was not granted.
      </h1>
      <p
        className="sw-NoWeb3_Description"
      >
        If a wallet extension is installed on your web browser, please verify that access to your Ethereum account has been granted and that it's available for this domain.
        <br />
        <strong>
          Need more help?
        </strong>
        <br />
        <a
          href="https://poanet.zendesk.com/hc/en-us/categories/360000586593-Wallets"
          target="blank"
        >
          Wallets HowTo
        </a>
         
        -
         
        <a
          href="https://github.com/poanetwork/token-wizard/wiki"
          target="blank"
        >
          Token Wizard Wiki
        </a>
      </p>
      <div
        className="sw-NoWeb3_ButtonsContainer"
      >
        <button
          className="btn-ButtonMetamask "
          onClick={[Function]}
          type="button"
        >
          <span
            className="btn-ButtonMetamask_Icon"
          />
          <span
            className="btn-ButtonMetamask_TextContainer"
          >
            <span
              className="btn-ButtonMetamask_TopText"
            >
              Download
            </span>
            <span
              className="btn-ButtonMetamask_BottomText"
            >
              Metamask
            </span>
          </span>
        </button>
        <button
          className="btn-ButtonNiftyWallet "
          onClick={[Function]}
          type="button"
        >
          <span
            className="btn-ButtonNiftyWallet_Icon"
          />
          <span
            className="btn-ButtonNiftyWallet_TextContainer"
          >
            <span
              className="btn-ButtonNiftyWallet_TopText"
            >
              Download
            </span>
            <span
              className="btn-ButtonNiftyWallet_BottomText"
            >
              Nifty Wallet
            </span>
          </span>
        </button>
        <a
          className="sw-NoWeb3_Button"
          href="/"
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</section>
`;
